python/res/enkf/jobs/scaling/job_config.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+import configsuite
+import six
+
+from copy import deepcopy
+
+from configsuite import MetaKeys as MK
+from configsuite import types
+
+
+@configsuite.validator_msg("Minimum length of index list must be > 1 for PCA")
+def _min_length(value):
+    return len(value) > 1
+
+
+@configsuite.validator_msg("Minimum value of index must be >= 0")
+def _min_value(value):
+    return value >= 0
+
+
+_num_convert_msg = "Will go through the input and try to convert to list of int"
+@configsuite.transformation_msg(_num_convert_msg)
+def _to_int_list(value):
+    value = deepcopy(value)
+    if isinstance(value, six.integer_types):
+        return [value]
+    elif isinstance(value, (list, tuple)):
+        value = ",".join([str(x) for x in value])
+    return _realize_list(value)
+
+
+@configsuite.transformation_msg("Convert ranges and singeltons into list")
+def _realize_list(input_string):
+    """If input_string is not a string, input_string will be returned. If input_string
+    is a string it is assumed to contain comma separated elements. Each element is
+    assumed to be either a singelton or a range. A singleton is a single number,
+    i.e.  7 or 14. A range is a lower and upper element of the range separated by a
+    single '-'. When provided with a string we will either return a list containing the
+    union of all the singeltons and the ranges, or raise a TypeError or ValueError if
+    it fails in the process. _realize_list('1,2,4-7,14-15') -> [1, 2, 4, 5, 6, 7, 14, 15]
+    """
```

Comment:
You've (re-)implemented a numparser here, but it does not accept non-positive integers, nor stepping.

Consider using https://pypi.org/project/pynumparser/ and do your own validation on the returned `IntSequence`.